src/Mod/Fem/Gui/TaskCreateElementSet.cpp

```diff
@@ -0,0 +1,820 @@
+        {
+            string xstr(startResultMesh.c_str());
+            createdMesh = newResultMesh.c_str();
+            Gui::Command::doCommand(Gui::Command::Doc, "obj1 = App.ActiveDocument.addObject('Fem::FemMeshObject', \'%s\')", startResultMesh.c_str());
+            Gui::Command::doCommand(Gui::Command::Doc, "Gui.ActiveDocument.getObject(\'%s\').Visibility = False", startResultMesh.c_str());
+            Gui::Command::doCommand(Gui::Command::Doc, "ne = Gui.ActiveDocument.getObject(\'%s\')", resultMesh.c_str());
+            Gui::Command::doCommand(Gui::Command::Doc, "obj1.FemMesh = ne.Object.FemMesh");
+        }
+        else if (passResult > 0)
+        {
+            createdMesh = newResultMesh.c_str();
+        }
+        passResult += 1;
+    }
+    else if (strcmp(meshType.c_str(), "femmesh") == 0)
+    {
+        if (passFemMesh == 0)
+        {
+            string xstr("no rename required");
+            createdMesh = newFemMesh.c_str();
+        }
+        else if (passFemMesh > 0)
+        {
+            createdMesh = newFemMesh.c_str();
+        }
+        passFemMesh += 1;
+    }
+    App::Document *doc = App::GetApplication().getActiveDocument();
+    uniqueMesh = doc->getUniqueObjectName(createdMesh.c_str());
+
+    Gui::Command::doCommand(Gui::Command::Doc, "Gui.ActiveDocument.getObject(\'%s\').Visibility = False", highLightMesh.c_str());
+    Gui::Command::doCommand(Gui::Command::Doc, "newermesh = Fem.read(\'%s\')", inp_file.c_str());
+    Gui::Command::doCommand(Gui::Command::Doc, "obj = App.ActiveDocument.addObject('Fem::FemMeshObject', \'%s\')", uniqueMesh.c_str());
+    Gui::Command::doCommand(Gui::Command::Doc, "obj.FemMesh = newermesh");
+    Gui::Command::doCommand(Gui::Command::Doc, "Gui.ActiveDocument.getObject(\'%s\').BackfaceCulling = False", uniqueMesh.c_str());
+    Gui::Command::doCommand(Gui::Command::Doc, "Gui.ActiveDocument.getObject(\'%s\').Visibility = True", uniqueMesh.c_str());
+
+    if (strcmp(meshType.c_str(), "result") == 0)
+    {
+        Gui::Command::doCommand(Gui::Command::Doc, "c = FreeCADGui.ActiveDocument.getObject(\'%s\')", uniqueMesh.c_str());
+        Gui::Command::doCommand(Gui::Command::Doc, "FreeCAD.ActiveDocument.%s.FemMesh = c.Object.FemMesh", resultMesh.c_str());
+        Gui::Command::doCommand(Gui::Command::Doc, "Gui.ActiveDocument.getObject(\'%s\').BackfaceCulling = False", resultMesh.c_str());
+        Gui::Command::doCommand(Gui::Command::Doc, "Gui.ActiveDocument.getObject(\'%s\').Visibility = True", resultMesh.c_str());
+        Gui::Command::doCommand(Gui::Command::Doc, "Gui.ActiveDocument.getObject(\'%s\').Visibility = False", uniqueMesh.c_str());
+    }
+} // generate mesh
+
+void writeToFile(std::string fileName, SMESHDS_Mesh *newMesh, int *nodeNumbers,
+                 double **nodeCoords, int rows, int requiredType)
+{
+    /*
+     */
+
+    std::map<int, std::string> elType2D;
+    std::map<int, std::string> elType3D;
+    elType2D[3] = "S3";
```

Comment:
Please use en enum here since numbers are hard to get later in the code. Also please keep the entries sorted for better overview